docker/docker_linux.go

```diff
@@ -317,3 +334,55 @@ func getCgroupMemFile(containerID, base, file string) (uint64, error) {
 	}
 	return strconv.ParseUint(lines[0], 10, 64)
 }
+
+// function to get the mount point of cgroup. by default it should be under /sys/fs/cgroup but
+// it could be mounted anywhere else if manually defined. Example cgroup entries in /proc/mounts would be
+//	 cgroup /sys/fs/cgroup/cpuset cgroup rw,relatime,cpuset 0 0
+//	 cgroup /sys/fs/cgroup/cpu cgroup rw,relatime,cpu 0 0
+//	 cgroup /sys/fs/cgroup/cpuacct cgroup rw,relatime,cpuacct 0 0
+//	 cgroup /sys/fs/cgroup/memory cgroup rw,relatime,memory 0 0
+//	 cgroup /sys/fs/cgroup/devices cgroup rw,relatime,devices 0 0
+//	 cgroup /sys/fs/cgroup/freezer cgroup rw,relatime,freezer 0 0
+//	 cgroup /sys/fs/cgroup/blkio cgroup rw,relatime,blkio 0 0
+//	 cgroup /sys/fs/cgroup/perf_event cgroup rw,relatime,perf_event 0 0
+//	 cgroup /sys/fs/cgroup/hugetlb cgroup rw,relatime,hugetlb 0 0
+// examples of target would be:
+// blkio  cpu  cpuacct  cpuset  devices  freezer  hugetlb  memory  net_cls  net_prio  perf_event  pids  systemd
+func getCgroupMountPoint(target string) (string, error) {
+	file, err := os.Open("/proc/mounts")
```

Comment:
is it worth caching this value just in memory? I assume the /proc/mounts would never change during a running process?